### DISCUSSION TITLE
Download release assets and zipball with CORS

### DISCUSSION THREAD

**Discussion by mastercoms** [community/community#45446] [Apps, API and Webhooks]
### Select Topic Area

Bug

### Body

Right now, unlike raw.githubusercontent.com, codeload.github.com (for downloading zipballs) and the release assets download bucket do not support CORS.

I need this functionality in order to implement web app based install methods, like through the File System Access API, or even enabling user customization of assets.

Of course, this would be enabled for public repositories only.

**Comment by AurelienRibon** [comment]
I am also interested in accessing a repo zipball through a web-app!

**Comment by khowling** [comment]
+1.  I create a release with an arm template built from bicep files in my repo, and I want to reference the release template download url in the `Deploy to Azure` button.  But hit this https://github.com/Azure/bicep/discussions/7756

**Comment by jschroeter** [comment]
I would also like to do this to provide firmware files for a web-based ESP flasher, see https://github.com/gdoor-org/gdoor/issues/12#issuecomment-2080207750

**Reply by gopro2027** [reply]
I now have this fully implemented on my repo https://github.com/gopro2027/ArduinoAirSuspensionController if you would like to take a look
Github actions to build the platformio esp32 images and create a release
Github pages html to load releases and flash them
Cloudflare worker reverse proxy to get past the cors issue

**Comment by puddly** [comment]
Same use case here.

We can hotlink to `raw.githubusercontent.com` but we cannot use GitHub releases with our [web flasher](https://skyconnect.home-assistant.io/firmware-update/): https://github.com/home-assistant/skyconnect.home-assistant.io/pull/105

**Comment by mrchantey** [comment]
Anyone know of an alternative service in the meantime?

**Comment by pzich** [comment]
I really hope this can get added. If anything it's odd that `raw.githubusercontent.com` *does* have CORS but there is no release/asset equivalent.

As a workaround, it seems like you could add an action that takes the release assets and commits them to a (probably different) repo, and then get the "raw" URL to *that*, but… 😬

**Reply by mrchantey** [reply]
> As a workaround, it seems like you could add an action that takes the release assets and commits them to a (probably different) repo

Yeah thats the approach I'm taking at the moment, here's the [workflow](https://github.com/mrchantey/beetmash/blob/4138e106b6391c33f78c03c46c75234b25abbe16/.github/workflows/release.yml) and this is what the commit step looks like:
```yaml
    - name: 🌐 Deploy to Pages Repo
      run: |
        cd ../beetmash-apps
        git config --global user.name "github-actions[bot]"
        git config --global user.email "github-actions[bot]@users.noreply.github.com"
        git add .
        git commit -m "Deploy from GitHub Actions"
        git push origin main
```

**Comment by gopro2027** [comment]
Bumping this because I just spend a week implementing github releases updates on my client firmware (as opposed to previously statically hosted firmwares in github pages), only for at the very last minute during updating my github pages flashing webpage to also support github releases, to get a cors issue.
I just built a whole suite revolving around using github releases... for releases... only for it to just be blocked by cors :/

**Reply by gopro2027** [reply]
> Bumping this because I just spend a week implementing github releases updates on my client firmware (as opposed to previously statically hosted firmwares in github pages), only for at the very last minute during updating my github pages flashing webpage to also support github releases, to get a cors issue. I just built a whole suite revolving around using github releases... for releases... only for it to just be blocked by cors :/

I have found a solution

Please create a [worker on Cloudflare](https://workers.cloudflare.com/) and use this code as a reverse proxy (Working example for downloading assets from releases by tag name). Cloudflare is free up to 100,000 requests per day.

```Javascript
/**
 * Reverse proxy for GitHub releases
 * Usage: https://WORKERNAME.CLOUDFLAREUSERNAME.workers.dev/?user=${user}&repo=${repo}&tag_name=${tag_name}&file_name=${file_name}
 */
export default {
  async fetch(request, env, ctx) {
      const url = new URL(request.url);
      const user = url.searchParams.get('user');
      const repo = url.searchParams.get('repo');
      const tag_name = url.searchParams.get('tag_name');
      const file_name = url.searchParams.get('file_name');
      
      if (!user || !repo || !tag_name || !file_name) {
        return new Response('Missing required parameters: user or repo or tag_name or file_name', {status: 400, statusText: 'Bad Request'});
      }

      return new Response(
        await (await fetch(`https://github.com/${user}/${repo}/releases/download/${tag_name}/${file_name}`)).arrayBuffer(),
        {
          headers: {
            "Access-Control-Allow-Headers": request.headers.get("Access-Control-Request-Headers"),
            "Access-Control-Allow-Methods": "GET,HEAD,POST,OPTIONS",
            "Access-Control-Allow-Origin": "*",
            "Access-Control-Max-Age": "300",
            "content-type": "application/octet-stream",
          }
        }
      );
  },
};
```

**Comment by sekedus** [comment]
Related: community/community#106849
